framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -158,21 +153,102 @@ private static Pattern anchored(String regex) {
+     */
+    private static Pair<String, String> parseMethodCall(
+            String sWithFormalParamNames, String s, ArrayList<String> argumentList) {
+        Expression e;
+        try {
+            e = parseExpression(sWithFormalParamNames);
+        } catch (ParseProblemException pe) {
+            return null;
+        }
+        if (e.getClass().equals(FieldAccessExpr.class)) {
+            FieldAccessExpr fieldAccessExpr = (FieldAccessExpr) e;
+            String fieldName = fieldAccessExpr.getName().toString();
+            if (s.indexOf(fieldName) == -1) fieldName = asFormalParameter(fieldName, argumentList);
+            String remaining = "." + fieldName;
+            String receiver = s.substring(0, s.length() - remaining.length());
+            Pair<Pair<String, String>, String> method = parseMethodCall(receiver, argumentList);
+            if (method != null) {
+                if (method.second.startsWith(".")) {
+                    String methodCall =
+                            method.first.first
+                                    + "("
+                                    + method.first.second
+                                    + ")."
+                                    + method.second.substring(1)
+                                    + remaining;
+                    if (methodCall.length() != s.length()) return null;
+                    return Pair.of(
+                            method.first.first + "(" + method.first.second + ")",
+                            method.second.substring(1) + remaining);
+                } else {
+                    String methodCall =
+                            method.first.first + "(" + method.first.second + ")." + fieldName;
+                    if (methodCall.length() != s.length()) return null;
+                    return Pair.of(method.first.first + "(" + method.first.second + ")", fieldName);
+                }
+            }
+            return null;
+        }
+        if (e.getClass().equals(MethodCallExpr.class)) {
+            Pair<Pair<String, String>, String> method = parseMethodCall(s, argumentList);
+            if (method != null && method.second.startsWith(".")) {
+                String methodCall =
+                        method.first.first
+                                + "("
+                                + method.first.second
+                                + ")."
+                                + method.second.substring(1);
+                if (methodCall.length() != s.length()) return null;
+                return Pair.of(
+                        method.first.first + "(" + method.first.second + ")",
+                        method.second.substring(1));
+            }
+        }
+        return null;
     }
 
     /**
      * Matches a field access. First of returned pair is object and second is field.
      *
      * @param s expression string
+     * @param context information about any receiver and arguments
```

Comment:
I don't understand this.  The method is parsing a field access (a "member select").  What are the arguments?  Please make the documentation descriptive.  Vague text like "any receiver and arguments" doesn't provide any information beyond what readers can determine from the type `FlowExpressionContext`.
Make the same fix elsewhere.